# HG changeset patch
# User Rik <rik@octave.org>
# Date 1533229319 25200
#      Thu Aug 02 10:01:59 2018 -0700
# Node ID 5bf1d8a0bcf7d0e2e10cfcd7e26f74d12dfad451
# Parent  a0923a7878e33a6ad1d58cadf520af1daf018dc5
isequalwithequalnans.m: Add to legacy list of functions (bug #50113).

* scripts/legacy/isequalwithequalnans.m: Script which directs users to function
isequaln.

* scripts/legacy/module.mk: Add isequalwithequalnans.m to build system.

diff --git a/scripts/legacy/isequalwithequalnans.m b/scripts/legacy/isequalwithequalnans.m
new file mode 100644
--- /dev/null
+++ b/scripts/legacy/isequalwithequalnans.m
@@ -0,0 +1,40 @@
+## Copyright (C) 2018 Rik Wehbring
+##
+## This file is part of Octave.
+##
+## Octave is free software: you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation, either version 3 of the License, or
+## (at your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <https://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {} isequalwithequalnans (@var{x1}, @var{x2}, @dots{})
+## This function is obsolete.  Use @code{isequaln} instead.
+## @seealso{isequaln}
+## @end deftypefn
+
+## At one time, Matlab docs stated that this function is obsolete and would be
+## removed in some future version.  Now users are told that it should be
+## avoided, but there is no mention of possible future removal.
+
+function retval = isequalwithequalnans (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:legacy-function",
+             "isequalwithequalnans is obsolete; please use isequaln instead");
+  endif
+
+  retval = isequaln (varargin{:});
+
+endfunction
diff --git a/scripts/legacy/module.mk b/scripts/legacy/module.mk
--- a/scripts/legacy/module.mk
+++ b/scripts/legacy/module.mk
@@ -1,11 +1,12 @@
 FCN_FILE_DIRS += scripts/legacy
 
 %canon_reldir%_FCN_FILES = \
+  %reldir%/isequalwithequalnans.m \
   %reldir%/isstr.m \
   %reldir%/setstr.m
 
 %canon_reldir%dir = $(fcnfiledir)/legacy
 
 %canon_reldir%_DATA = $(%canon_reldir%_FCN_FILES)
 
 FCN_FILES += $(%canon_reldir%_FCN_FILES)
